src/include/miopen/sqlite_db.hpp

```diff
@@ -278,38 +282,50 @@ class SQLiteBase
         return AllFound;
     }
     template <typename... U>
-    inline auto FindRecord(U&... args)
+    inline boost::optional<DbRecord> FindRecord(U&... args)
     {
+        if(DisableDbFileIO)
```

Comment:
This also affects the SQLite-based kernel cache `KernDb : public SQLiteBase<KernDb>`, but it shouldn't

Reply:
Hmm... Now I am not sure, see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/280#issuecomment-644931011